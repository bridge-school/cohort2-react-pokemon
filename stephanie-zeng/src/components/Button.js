import React from 'react';

export const Button = ({ children, submitSearchTerm }) => <button onClick={submitSearchTerm}>{children}</button>;
