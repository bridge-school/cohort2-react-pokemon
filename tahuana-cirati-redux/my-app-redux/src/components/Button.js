import React, { Component } from 'react';
import { connect } from 'react-redux';
import fetchPokemonAction from '../actions/button-actions';


class Button extends Component {
    onClickSearch() {
        this.props.onClickSearch(this.props.inputSearch)
    }

    render() {
        return (
            <input className="Button" type='button' value='Search' onClick={() => this.onClickSearch()} />
        )
    }
}

const mapStateToProps = store => { 
    return ({
    inputSearch: store.inputSearch
  })}
  
const mapDispatchToProps = {
    onClickSearch: fetchPokemonAction
  } 
  
export default connect(mapStateToProps, mapDispatchToProps)(Button);




