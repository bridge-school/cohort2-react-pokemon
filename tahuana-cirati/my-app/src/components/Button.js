import React from 'react';

const Button = ( {fetchPokemon} ) => (
    <input className="Button" type='button' value='Search' onClick={() => fetchPokemon()} />
);

export default Button;

