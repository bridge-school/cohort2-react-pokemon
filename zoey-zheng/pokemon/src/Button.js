import React from 'react';

export const Button = ({
  fetchData,
  inputValue,
  clearInput,
  updateIsFetching
}) => {
  const handleOnClick = event => {
    event.preventDefault();
    if (!inputValue) {
      alert('oops! please enter a Pokemon name!');
    } else {
      updateIsFetching();
      fetchData(inputValue);
    }
    clearInput();
  };

  return <button onClick={handleOnClick}> Fetch </button>;
};
