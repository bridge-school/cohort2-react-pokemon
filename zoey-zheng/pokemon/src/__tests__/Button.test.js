import React from 'react';
import { Button } from '../Button.js';
import renderer from 'react-test-renderer';

describe('button component tests', () => {
  it('renders button component', () => {
    const button = renderer.create(<Button />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
