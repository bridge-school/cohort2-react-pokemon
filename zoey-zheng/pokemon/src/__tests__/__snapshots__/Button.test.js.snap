// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component tests renders button component 1`] = `
<button
  onClick={[Function]}
>
   Submit 
</button>
`;
